### PR TITLE
Data Explorer - styling - demo blockers

### LINKED ISSUE #1663: Implement design for Exploration Editor
- Blocked by #1656

### LINKED ISSUE #1795: Implement automatic list aggregation when joining on one-to-many link
@seancolsen reports:

> The process of joining on a reverse relationship might be confusing. Set base table to Authors. Add id, first name, last name. Observe 14 results. Ok cool. Now add "Publications"."Title" and observe 65 results. What? I understand what happened, but someone who has never used SQL is likely to be confused by this jump in results. 
>
> I have a vague memory of expressing this concern many months ago and hearing from someone that our design was such that one-to-many data like this would be pulled in with an automatic array aggregation.

The [spec](https://wiki.mathesar.org/en/design/specs/visual-query-builder) did specify automatic aggregation, but this wasn't implemented per [this email thread](https://groups.google.com/a/mathesar.org/g/core-team/c/qR-vcl88Uac/m/jsK5QEhFBAAJ) to reduce scope. This is important for good UX and we should implement it for the live demo.

### PR DESCRIPTION
**Pull request by pavish**
Related to #2036 
Fixes #1795 
Fixes #1663 
Related to #1664 

### This PR:
* Shows auto summarization modal and performs auto-summarization when user clicks on a column with one-to-many link
* Adds save & close dropdown option to save button
* Shows save status indicator when unsaved changes are present (in exploration editor)
* Shows initial help tutorial when there are no queries in current schema
* Shows array icon in left pane for columns with one-to-many links
* Adds Hide transformation
  - Note: The frontend for it is implemented here, but there's a backend bug which results in a 500 when a hide transformation is added. It is handed in https://github.com/centerofci/mathesar/pull/2083.
* Formats array cell content based on display_options.
   - Treats array cell as a compound cell type
* It also contains some performance improvements:
  - Previously display formatter object was created once per cell for number and money columns. This PR changes them to be created only once per column.

### Note:
* This PR does not style the modals
  - Save exploration modal
  - Confirmation modal for auto-summarization
* Reasons:
   - These modals are common throughout the product, and any styling work should be done & verified throughout. So, I'm not considering them specific to Data Explorer.
   - I noticed an open issue https://github.com/centerofci/mathesar/issues/2085 to change modal background color. We'd probably want to update modal header font sizes as part of the same issue or a different one.
   - I did not do any work here so as to not clash with others working on the same styling issue.

@kgodey This PR handles everything needed for the demo video from Data Explorer (except modal styling - see note above). Let me know if there's anything else thats blocking the demo video in DE, after this PR is merged.

With this PR merged, I'd like to close #1663 and handle the remaining items in #1664 

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kgodey** [comment]
Okay, thanks @pavish! @seancolsen are you okay with taking on modal styling work?

**Review by seancolsen — APPROVED** [review]
@pavish this works well.

- Thanks for pushing a8027a7005384482cfe6a0c23344079b790117f9. That alleviates some confusion.

- I opened the following tickets with issues that can be addressed later

    - #2093
    - #2094

- Really smart, moving the display formatting logic up to the column instead of the cell!

- I made some comments on the code but nothing is pressing. Feel free to merge this yourself once fixing the conflicts.

File: mathesar_ui/src/systems/data-explorer/input-sidebar/transformations-pane/TransformationsPane.svelte
Comment (pavish) [reply mathesar_ui/src/systems/data-explorer/input-sidebar/transformations-pane/TransformationsPane.svelte:160]: They aren't present on the figma designs and we do not have proper icons that represent each operation for the options listed here. When we have the required custom icons , we can add them.

File: mathesar_ui/src/systems/data-explorer/ActionsPane.svelte
Comment (pavish) [reply mathesar_ui/src/systems/data-explorer/ActionsPane.svelte]: I agree. Fixed here: https://github.com/centerofci/mathesar/pull/2091/commits/287e94c53d059a82207a4fb48494d18e2aa7a5d4

File: mathesar_ui/src/systems/data-explorer/DataExplorer.svelte
Comment (pavish) [reply mathesar_ui/src/systems/data-explorer/DataExplorer.svelte]: Fixed here: https://github.com/centerofci/mathesar/pull/2091/commits/8ba2d8eb04a33ee3f4af8bc8429f02df3ea625f1

File: mathesar_ui/src/systems/data-explorer/QueryHideTransformationModel.ts
Comment (pavish) [reply mathesar_ui/src/systems/data-explorer/QueryHideTransformationModel.ts]: Fixed here: https://github.com/centerofci/mathesar/pull/2091/commits/23b3541e12ae32b4d18ba4d9a3eaa1e98caef613

**Comment by pavish** [comment]
@seancolsen Since the changes added for the review comments are quite small and I already have your approval, I'm going to go ahead and merge this in. Feel free to take a lost post merge.

**Comment by seancolsen** [comment]
@kgodey 

> @seancolsen are you okay with taking on modal styling work?

Yes, the only think left to do for these modals is change the background color, which is captured in #2085 (already assigned to me).

**Pull request merged**
